Wikipedia/Code/ExploreViewController.swift

```diff
@@ -1126,13 +1126,28 @@ extension ExploreViewController: WKImageRecommendationsDelegate {
         navigationController?.pushViewController(articleViewController, animated: true)
     }
 
-    func imageRecommendationsUserDidTapInsertImage(with imageData: WKImageRecommendationsViewModel.WKImageRecommendationData) {
+    func imageRecommendationsUserDidTapInsertImage(project: WKData.WKProject, title: String, with imageData: WKImageRecommendationsViewModel.WKImageRecommendationData) {
+
+        var components = URLComponents()
+        components.scheme = "https"
+
+        switch project {
+        case .wikipedia(let language):
+            components.host = "\(language.languageCode).wikipedia.org"
+        default:
+            assertionFailure("Unexpected project for image recommendations")
+        }
```

Comment:
I recently made a WKProject.siteURL convenience method that will save you some lines here.